test/gtest/uct/test_peer_failure.cc

```diff
@@ -111,9 +113,15 @@ class test_uct_peer_failure : public uct_test {
         m_entities.push_back(m_receivers.back());
         m_sender->connect(m_receivers.size() - 1, *m_receivers.back(), 0);
 
-        am_handler_setter(this)(m_receivers.back());
-        /* Make sure that TL is up and has resources */
-        send_recv_am(m_receivers.size() - 1);
+        uct_iface_attr_t iface_attr;
+        ucs_status_t status = uct_iface_query(m_sender->ep(0)->iface,
+                                              &iface_attr);
+        ASSERT_UCS_OK(status);
+        if (iface_attr.cap.flags & UCT_IFACE_FLAG_AM_SHORT) {
+            /* Make sure that TL is up and has resources */
+            am_handler_setter(this)(m_receivers.back());
+            send_recv_am(m_receivers.size() - 1);
+        }
```

Comment:
is there a need to add check for others?

Reply:
may be, but later